plugins/analytics-module-ga/src/apis/implementations/AnalyticsApi/GoogleAnalytics.ts

```diff
@@ -154,6 +166,17 @@ export class GoogleAnalytics implements AnalyticsApi {
       return;
     }
 
+    if (this.virtualSearchPageView.mode !== 'disabled' && action === 'search') {
+      const { mountPath, queryParam } = this.virtualSearchPageView;
+      this.capture.pageview(
+        `${mountPath}?${queryParam}=${subject}`,
```

Comment:
I believe GA Site Search also a concept of site search _categories_ that might map well to the `types` (provided contextually as `searchTypes` [here](https://github.com/backstage/backstage/blob/master/plugins/search-react/src/context/SearchContext.tsx#L196-L202)).  Could be worth exposing that here too.

Reply:
I assume it should be configurable in a way similar to `queryParam`? 

For example, if `categoryQuery` is present in config, then the pageview is sent to `${mountPath}?${queryParam}=${subject}&${categoryQuery}=${context.searchTypes ?? ''}`, otherwise no category is attached. Now that I think about it, probably better to rename `queryParam` to something like `searchQuery`. What do you think?

Reply:
Yeah, that makes sense to me.

Although now that I think about it, I wonder if either of these should be configurable?  If the intention is to get the virtual page view to match a real page view, then it makes sense to be able to configure the path (since that can be configured by changing where the main search page is mounted)...  But I think both of these are implementation details of the search page itself and should always be `query` and `types[]`, as in https://demo.backstage.io/search?query=t&types%5B%5D=techdocs

...I suppose the exception would be if you have a search experience that is not managed by the search plugin?

Reply:
My first instinct was to make it always match the actual `/search` path, but then I realized that that would mean having duplicated pageviews when searching on the main search page (instead of the sidebar search). That's why I decided to make it configurable, and currently we set it up in such a way that "real" pageviews on the main search page just get ignored by Site Search, and only virtual pageviews get parsed (based on the configurable query parameter), and we also can separate duplicating pageviews from real ones because of the configurable path.

Reply:
I've added the support for search categories, please take a look